src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetInvitations/GetInvitationsQueryHandlerTests.cs

```diff
@@ -0,0 +1,582 @@
+            {
+                var query = new GetInvitationsQuery(_projectName2, null, filter);
+                var dut = new GetInvitationsQueryHandler(context);
+
+                var result = await dut.Handle(query, default);
+                AssertCount(result.Data, 1);
+            }
+        }
+
+        [TestMethod]
+        public async Task Handler_ShouldFilterOnCommPkgNoStartWith()
+        {
+            var filter = new Filter { CommPkgNoStartsWith = _commPkgNo2 };
+
+            using (var context =
+                new IPOContext(_dbContextOptions, _plantProvider, _eventDispatcher, _currentUserProvider))
+            {
+                var query = new GetInvitationsQuery(_projectName2, null, filter);
+                var dut = new GetInvitationsQueryHandler(context);
+
+                var result = await dut.Handle(query, default);
+                AssertCount(result.Data, 1);
+            }
+        }
+
+        [TestMethod]
+        public async Task Handler_ShouldFilterOnCommPkgNoStartWith2()
+        {
+            var filter = new Filter { CommPkgNoStartsWith = _commPkgNo };
+
+            using (var context =
+                new IPOContext(_dbContextOptions, _plantProvider, _eventDispatcher, _currentUserProvider))
+            {
+                var query = new GetInvitationsQuery(_projectName2, null, filter);
+                var dut = new GetInvitationsQueryHandler(context);
+
+                var result = await dut.Handle(query, default);
+                AssertCount(result.Data, 2);
+            }
+        }
+
+        [TestMethod]
+        public async Task Handler_ShouldFilterOnMcPkgNoStartWith()
+        {
+            var filter = new Filter { McPkgNoStartsWith = _mcPkgNo };
+
+            using (var context =
+                new IPOContext(_dbContextOptions, _plantProvider, _eventDispatcher, _currentUserProvider))
+            {
+                var query = new GetInvitationsQuery(_projectName2, null, filter);
+                var dut = new GetInvitationsQueryHandler(context);
+
+                var result = await dut.Handle(query, default);
+                AssertCount(result.Data, 1);
+            }
+        }
+
+        [TestMethod]
+        public async Task Handler_ShouldFilterOnPunchOutOverdue()
+        {
+            var filter = new Filter { PunchOutDates = new List<PunchOutDateFilterType> { PunchOutDateFilterType.Overdue } };
+
+            using (var context =
+                new IPOContext(_dbContextOptions, _plantProvider, _eventDispatcher, _currentUserProvider))
+            {
+                var query = new GetInvitationsQuery(_projectName2, null, filter);
+                var dut = new GetInvitationsQueryHandler(context);
+
+                var result = await dut.Handle(query, default);
+                AssertCount(result.Data, 2);
+            }
+        }
+
+        [TestMethod]
+        public async Task Handler_ShouldFilterOnPunchOutInNextWeek()
```

Comment:
Consider to duplicate this test, but use TimeService to move the time backward so filtering on NextWeek can end with AssertCount(result.Data, 1);